### DISCUSSION TITLE
MQTT for microPython/Python

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#12733] [General]
I've been using the MQTT import "umqttsimple" quite successfully in a ESP32 application, and I'm currently using "paho.mqtt" in standard Python. Does anyone know whether "umqttsimple" would be compatible with standard Python?

MM

**Comment by nastiliano** [comment]
Hello @mmliam ,

Just right now I wanted to ask the same to people.
Now I want to bring Internet connectivity to my device project and my doubt to start to research is: what is the "standard" MQTT library in MicroPython.

I read that in "Python" the most used library is "paho.mqtt" and I wanted to know which is the most used in "MicroPython" to syncro the job and start to study how to send packets over Internet like "Open Door", etc.
My app works by WiFi very good and now I want to connect the ESP32 device to "MQTT" server to facility the "user" configuration work and not have to put a public IP with router opened specific port, etc.

Thanks.

**Comment by beetlegigg** [comment]
Have a peep and the mqtt-as.py from Peter Hinch.  It has the advantage of being an asyncio implementation and its worth a consideration for that aspect, as well as being a robust mqtt client.  You can read all about it at https://github.com/peterhinch/micropython-mqtt

**Comment by bidrohini** [comment]
"umqttsimple" is not suitable for standard python. For standard python, you have to use  "paho.mqtt" library.

**Comment by Anyel-ec** [comment]
Unfortunately, "umqttsimple" is a MicroPython-specific library designed for IoT devices like the ESP32. It is not directly supported by standard Python, as it is tailored for the MicroPython environment and limitations. To work with MQTT in standard Python, it is best to use libraries such as "paho.mqtt", which are designed for use in this environment. 😊

**Comment by nastiliano** [comment]
Thanks to everybody.

I will start using "paho.mqtt" in my PC/mobile application and in my ESP32 I will use @peterhinch "async MQTT" library.

Thanks again. Best regards.

**Comment by camp-easy** [comment]
hello everyone, im trying to use this code to connect to mqtt server:
try:
    from umqtt import simple
except ImportError:
    import mip
    mip.install("micropython-umqtt.simple")
    from umqtt import simple


but i receive this error

OSError: [Errno 110] ETIMEDOUT,
does anyone know some solutions? a better code? thank you in advnce for the support

**Comment by peterhinch** [comment]
```py
>>> mip.install("umqtt.simple")
Installing umqtt.simple (latest) from https://micropython.org/pi/v2 to /lib
Copying: /lib/umqtt/simple.mpy
Done
>>> from umqtt import simple
>>> 
```

**Reply by ctimmer** [reply]
I get this result from your example:
```
MicroPython v1.21.0 on 2023-10-06; Raspberry Pi Pico W with RP2040
Type "help()" for more information.

>>> import mip
>>> mip.install(“umqtt.simple”)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name '“umqtt' isn't defined

>>> mip.install('umqtt.simple')
Installing umqtt.simple (latest) from https://micropython.org/pi/v2 to /lib
Copying: /lib/umqtt/simple.mpy
Done
>>> 
```
What am I missing?

Curt

**Reply by robert-hh** [reply]
The “ character in the first attempt is not a double quote character " with the hex code 0x22. It just looks similar and has the utf-8 code 0xe2 0x80 0x9c.
